packages/flutter/test/widgets/system_context_menu_test.dart

```diff
@@ -895,4 +895,57 @@ void main() {
     },
     skip: kIsWeb, // [intended] SystemContextMenu is not supported on web.
   );
+
+  testWidgets('Default iOS SystemContextMenu includes Share for non-empty selection',
+      (WidgetTester tester) async {
+    final controller = TextEditingController(text: 'Hello world');
+
+    // Force iOS + tell Flutter we support the iOS system context menu path.
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+    await tester.pumpWidget(
+      MediaQuery(
+        data: const MediaQueryData(supportsShowingSystemContextMenu: true),
+        child: const Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: _HostEditable(), // minimal EditableText wrapper
+          ),
+        ),
+      ),
+    );
+
+    // Setup selection so sharing is meaningful.
+    final state = tester.state<EditableTextState>(find.byType(EditableText));
+    state.widget.controller.text = controller.text;
+    state.widget.controller.selection =
+        const TextSelection(baseOffset: 0, extentOffset: 5); // "Hello"
+    state.widget.focusNode.requestFocus();
+    await tester.pump();
+
+    // Get iOS SYSTEM menu defaults and assert Share is present.
+    final items = SystemContextMenu.getDefaultItems(state);
+    expect(items.any((e) => e is IOSSystemContextMenuItemShare), isTrue);
+
+    debugDefaultTargetPlatformOverride = null;
+  });
```

Comment:
How I would test this: Listen to 'ContextMenu.showSystemContextMenu' (see elsewhere in the file for how to do that), then show the context menu and expect that it was called with a "share" item.